aten/src/ATen/native/vulkan/glsl/reflection_pad2d.glsl

```diff
@@ -0,0 +1,31 @@
+#version 450 core
+#define PRECISION $precision
+
+layout(std430) buffer;
+
+/* Qualifiers: layout - storage - precision - memory */
+
+layout(set = 0, binding = 0) uniform PRECISION restrict writeonly image3D   uOutput;
+layout(set = 0, binding = 1) uniform PRECISION                    sampler3D uInput;
+layout(set = 0, binding = 2) uniform PRECISION restrict           Block {
+  ivec4 size;
+  ivec4 pad;
+} uBlock;
+
+layout(local_size_x_id = 0, local_size_y_id = 1, local_size_z_id = 2) in;
+
+void main() {
+  const ivec3 pos = ivec3(gl_GlobalInvocationID);
+
+  if (all(lessThan(pos, uBlock.size.xyz))) {
+    const int xoff_pre  = 2*max(uBlock.pad.x - pos.x, 0);
+    const int xoff_post = 2*max(pos.x - (uBlock.size.x - 1 - uBlock.pad.y), 0);
+    const int yoff_pre  = 2*max(uBlock.pad.z - pos.y, 0);
+    const int yoff_post = 2*max(pos.y - (uBlock.size.y - 1 - uBlock.pad.w), 0);
```

Comment:
I think these statements can be packed in terms of vectors as opposed to scalars.  For instance

```
const int2 xoff = 2 * max(uBlock.pad.xz - pos.xy, 0);
const int2 yoff = 2 * max(pos.xy - (uBlock.size.xy - 1 - uBlock.pad.yw), 0);
```

and even then you might be able to refactor it (with some thought - not obvious to me now) to

```
const int4 offset = 2 * max( ... );
```

Reply:
you're absolutely right, can definitely make this part more compact.